posthog/models/messaging.py

```diff
@@ -0,0 +1,29 @@
+import hashlib
+
+from django.conf import settings
+from django.db import models
+
+from .utils import UUIDModel
+
+
+class MessagingRecordManager(models.Manager):
+    def get_or_create(self, defaults=None, **kwargs):
+        raw_email = kwargs.pop("raw_email", None)
+
+        if raw_email:
+            kwargs["email_hash"] = hashlib.sha256(f"{settings.SECRET_KEY}_{raw_email}".encode()).hexdigest()
```

Comment:
I don't quite get why the actual email address is destroyed by hashing here. IMO it could actually come in handy to know who in particular wasn't reached by emailing due to a bug or something.

Reply:
The reasoning behind this is for privacy-preservation reasons. Having the email here would require extra effort to maintain it so that PII is cleared when it has to. This way, even if user data must be permanently deleted from the system, the model will not contain any. I guess if you were trying to debug some delivery issue and you know the email address you could compute the hash and look up using the hash. POV?

Reply:
Hm, that was also my reasoning for keeping messaging records tied to users, they could get cascade deleted then.